test/test-experiment-recipe-firefox.ts

```diff
@@ -0,0 +1,45 @@
+import {typeGuards} from "..";
+
+const TEST_EXPERIMENT = {
+  "id": "ABOUTWELCOME-PULL-FACTOR-REINFORCEMENT-76-RELEASE",
+  "enabled": true,
+  "filter_expression": "(env.version >= '76.' && env.version < '77.' && env.channel == 'release' && !(env.telemetry.main.environment.profile.creationDate < ('2020-05-13'|date / 1000 / 60 / 60 / 24))) || (locale == 'en-US' && [userId, \"aboutwelcome-pull-factor-reinforcement-76-release\"]|bucketSample(0, 2000, 10000) && (!('trailhead.firstrun.didSeeAboutWelcome'|preferenceValue) || 'bug-1637316-message-aboutwelcome-pull-factor-reinforcement-76-rel-release-76-77' in activeExperiments))",
```

Comment:
Q: Do we ever validate that this is a valid string/JEXL expression (and it isn't missing a quote or parenthesis), or is that something that Experimenter does downstream? Or will the `filter_expression` will always be valid since it isn't being provided by the user and is something that we're generating from Experimenter and the intake form.
/cc @jaredlockhart @mythmon 

I'm not sure if I fully understand our TypeScript 👉 AJV 👉 schema validation flow.

Reply:
"is a valid JEXL expression" is not something that either Typescript or JSON schema can express. All we can say is that this is a string, and that is what the type system says. This means that consumers of this data must be prepared to handling invalid expressions. This promise is nicely expressed by using the type `string` instead of some JEXL-specific type.

No one should really be introspecting these strings anyways, except to execute them in a JEXL engine. If we get to the point where we need to determine the contents of a filter, we really ought to be using a structured format like Normandy's filter objects.